### PR TITLE
Simplify not adding type header in SqsTemplate sent messages (#659)

### PR DESCRIPTION
**Pull request by levys17**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

- Implemented the method `doNotSendTypeHeader()` in `AbstractMessagingMessageConverter` to simplify the way to not send type information in messages sent with `SqsTemplate`.

## :bulb: Motivation and Context
Issue related: #659 

## :green_heart: How did you test it?
Updated unit test `shouldSendAndReceiveRecordMessageWithoutPayloadInfoHeader` in `SqsTemplateIntegrationTests` with this new implementation instead of old one.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] I updated reference documentation to reflect the change
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Review by tomazfernandes — COMMENTED** [review]
Hey @levys17, implementation looks good!

Can you please add this option to the documentation and include a unit test?

Thanks!

**Comment by tomazfernandes** [comment]
> Hey @levys17, implementation looks good!
> 
> Can you please add this option to the documentation and include a unit test?
> 
> Thanks!

Oh, sorry, nevermind about the test, now I see we already had one.

So if you can just add it to the docs please - let me know if you need any help with that.

**Comment by levys17** [comment]
> > Hey @levys17, implementation looks good!
> > Can you please add this option to the documentation and include a unit test?
> > Thanks!
> 
> Oh, sorry, nevermind about the test, now I see we already had one.
> 
> So if you can just add it to the docs please - let me know if you need any help with that.

Hey, @tomazfernandes ! Do you think it's ok?
image

File: spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/support/converter/AbstractMessagingMessageConverter.java [hunk 1]
```diff
@@ -145,6 +145,13 @@ public void setHeaderMapper(HeaderMapper<S> headerMapper) {
 		this.headerMapper = headerMapper;
 	}
 
+	/**
+	 * Configure to not send TypeHeader information for {@link software.amazon.awssdk.services.sqs.model.Message} instances.
```
Comment (tomazfernandes) [inline spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/support/converter/AbstractMessagingMessageConverter.java:149]: On a second thought, how about something like:

`Configure the converter to not include payload type information in the message headers.`?

**Comment by tomazfernandes** [comment]
> Hey, @tomazfernandes ! Do you think it's ok?

Hey @levys17! That's about it, but please commit the text to the PR so it's easier to review if you can.

File: spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/support/converter/AbstractMessagingMessageConverter.java [hunk 2]
```diff
@@ -145,6 +145,13 @@ public void setHeaderMapper(HeaderMapper<S> headerMapper) {
 		this.headerMapper = headerMapper;
 	}
 
+	/**
+	 * Configure the converter to not include payload type information in the {@link software.amazon.awssdk.services.sqs.model.Message} headers.
+	 */
+	public void doNotSendTypeHeader() {
```
Comment (tomazfernandes) [inline spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/support/converter/AbstractMessagingMessageConverter.java:151]: Second thoughts again, sorry - how about `doNotSendPayloadTypeHeader`?

File: docs/src/main/asciidoc/sqs.adoc [hunk 3]
```diff
@@ -1231,6 +1231,8 @@ public interface MessagingMessageConverter<S> {
 
 The default header-based type mapping can be configured to use a different header name by using the `setPayloadTypeHeader` method.
 
+Also is possible to not send `TypeHeader` by using the `doNotSendTypeHeader` method.
```
Comment (tomazfernandes) [inline docs/src/main/asciidoc/sqs.adoc:1234]: We can use a similar approach to the javadocs here, something like:

"It is also possible not include payload type information in the header by using..."

**Review by tomazfernandes — COMMENTED** [review]
Hey @levys17, thanks for the changes, we're almost there.

Left a couple of comments.

**Comment by levys17** [comment]
Hey, @tomazfernandes ! Thanks for the review! Can you check again?

File: docs/src/main/asciidoc/sqs.adoc [hunk 4]
```diff
@@ -1231,6 +1231,8 @@ public interface MessagingMessageConverter<S> {
 
 The default header-based type mapping can be configured to use a different header name by using the `setPayloadTypeHeader` method.
 
+It is also possible not include payload type information in the header by using `doNotSendPayloadTypeHeader` method.
```
Comment (tomazfernandes) [inline docs/src/main/asciidoc/sqs.adoc:1234]: Just two minor adjustments:
"It is also possible not to include"
and
"by using the ... method"

File: spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/integration/SqsTemplateIntegrationTests.java [hunk 5]
```diff
@@ -305,7 +305,7 @@ void shouldSendAndReceiveBatchFifo() {
 	@Test
 	void shouldSendAndReceiveRecordMessageWithoutPayloadInfoHeader() {
 		SqsTemplate template = SqsTemplate.builder().sqsAsyncClient(this.asyncClient)
-				.configureDefaultConverter(converter -> converter.setPayloadTypeHeaderValueFunction(msg -> null))
+				.configureDefaultConverter(converter -> converter.doNotSendTypeHeader())
```
Comment (tomazfernandes) [inline spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/integration/SqsTemplateIntegrationTests.java:308]: Something seems off here - shouldn't this be `converter.doNotSendTypeHeader()`?

Comment (tomazfernandes) [reply spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/integration/SqsTemplateIntegrationTests.java:308]: Yup, CI broke. @levys17, if you can please fix the test. Thanks!

**Pull request merged**